horace_core/algorithms/cut.m

```diff
@@ -0,0 +1,30 @@
+function wout = cut(source, varargin)
+%%CUT Take a cut from the given data source.
+%
+% Inputs:
+% -------
+% source     An `sqw` or `dnd` object or a path to a valid .sqw or .dnd file.
+%
+% For more help see sqw/cut.
+%
+DND_CLASSES = {'d0d', 'd1d', 'd2d', 'd3d', 'd4d'};
+
+if is_string(source)
+    ldr = sqw_formats_factory.instance().get_loader(source);
+    if ldr.sqw_type
+        pixel_page_size = get(hor_config, 'pixel_page_size');
+        sqw_dnd_obj = sqw(source, 'pixel_page_size', pixel_page_size);
```

Comment:
My problem here was that you don't necessarily know what kind of file the input is. It could be a `.sqw` file containing pixels, or an `sqw` with no pixels, or it could be a `.d2d` containing `dnd` data (I wasn't actually aware these existed, but there appears to be a test that uses one). Using the loader allows us to ID what kind of object we're dealing with.

The reason we don't use the loader to read our `sqw` object is because we need to pass the `pixel_page_size` argument to the constructor - there's no way to do this through the loader.

Reply:
Maybe it is worth commenting the code with this information